content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -252,21 +254,553 @@ cover more specific details on anonymous access types.
+objects (of anonymous access types) to a named type that has the same
+designated subtype. For example, if we have an access object :ada:`IA` of
+an anonymous :ada:`access Integer` type, we can convert it to the named
+:ada:`Integer_Access` type, provided this named access type is compatible
+with the anonymous access type, e.g.:
+
+.. code-block:: ada
+
+    type Integer_Access is access all Integer
+
+Let's see a complete code example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Deallocation_Anonymous_Access_To_Object_Erronoeus
+
+    with Ada.Unchecked_Deallocation;
+
+    procedure Show_Dangerous_Deallocation is
+       type Integer_Access is
+          access all Integer;
+
+       procedure Free is
+         new Ada.Unchecked_Deallocation
+           (Object => Integer,
+            Name   => Integer_Access);
+
+       IA : access Integer;
+    begin
+       IA := new Integer;
+       IA.all := 30;
+
+       --  Potentially erroneous deallocation via type
+       --  conversion:
+       Free (Integer_Access (IA));
+
+    end Show_Dangerous_Deallocation;
+
+This example declares the :ada:`IA` access object of the anonymous
+:ada:`access Integer` type. After allocating an object for :ada:`IA` via
+:ada:`new`, we try to deallocate it by first converting it to the
+:ada:`Integer_Access` type, so that we can call the :ada:`Free` procedure to
+actually deallocate the object. Although this code compiles, it'll only work
+if both :ada:`access Integer` and :ada:`Integer_Access` types are using the
+same memory pool. Since we cannot really determine this, the result is
+potentially erroneous: it'll work if the compiler selected the same pool, but
+it'll fail otherwise.
+
+.. admonition:: Important
+
+    Because allocating memory for anonymous access types is potentially
+    dangerous, we can use the :ada:`No_Anonymous_Allocators` restriction
+    |mdash| which is available since Ada 2012 |mdash| to prevent this kind
+    of memory allocation being used in the code. For example:
+
+    .. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.No_Anonymous_Allocators
+        :class: ada-expect-compile-error
+
+        pragma Restrictions (No_Anonymous_Allocators);
+
+        procedure Show_Dangerous_Allocation is
+           IA : access Integer;
+        begin
+           IA := new Integer;
+           IA.all := 30;
+        end Show_Dangerous_Allocation;
+
+Possible solution using named access types
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A better solution to avoid issues when allocating and deallocating memory
+for anonymous access-to-object types is to allocate the object using a known
+pool. As mentioned before, the memory pool associated with a named access
+type is well-defined, so we can use this kind of types for memory allocation.
```

Comment:
"this kind of types" should probably be "these kinds of types"

Reply:
Ok, I'll change it, but I'm not sure if that would be correct. In fact, I was trying to imply that named access types is *one* kind of types.

Reply:
OK, I see. In that case it should be "this kind of type" ie "kind" and "type" need to agree regarding the tense.

Reply:
Good, I´ll revert my last change. :-)